### ISSUE TITLE
Transport responding timeouts in V2 not working

### ISSUE THREAD

**Issue by jrbalsas** [containous/traefik#5763]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Configure transport responding timeouts  in static config file or from command line:

```yaml
entrypoints:
  web:
    address: ":80"
    transport:
      respondingTimeouts: 
        readTimeout: "10s"
        idleTimeout: 10
```
or with CLI options
```
--entryPoints.web.address=":80" \
--entryPoints.web.transport.respondingTimeouts.readTimeout="10s" \
--entryPoints.web.transport.respondingTimeouts.idleTimeout=10 \
```

### What did you expect to see?

After keeping an open connection it should be closed after the configured timeout, e.g.
``` 
$ time nc localhost 80
```

### What did you see instead?

Traefik does not close the connection after waiting for configured time

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      2.0.4
Codename:     montdor
Go version:   go1.13.3
Built:        2019-10-28T20:23:57Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```yaml
log:
  level: DEBUG
providers:
  file:
    filename: /etc/traefik/dynamic-conf.yml
entrypoints:
  web:
    address: ":80"
    transport:
      respondingTimeouts:
        readTimeout: "10s"
        writeTimeout: "10s"
        idleTimeout: 10

```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

```
traefik_1    | time="2019-11-01T13:36:38Z" level=info msg="Configuration loaded from file: /etc/traefik/traefik.yml"
traefik_1    | time="2019-11-01T13:36:38Z" level=info msg="Traefik version 2.0.4 built on 2019-10-28T20:23:57Z"
traefik_1    | time="2019-11-01T13:36:38Z" level=debug msg="Static configuration loaded {\"global\":{\"checkNewVersion\":true},\"serversTransport\":{\"maxIdleConnsPerHost\":200},\"entryPoints\":{\"web\":{\"address\":\":80\",\"transport\":{\"lifeCycle\":{\"graceTimeOut\":10000000000},\"respondingTimeouts\":{\"readTimeout\":10000000000,\"writeTimeout\":10000000000,\"idleTimeout\":10000000000}},\"forwardedHeaders\":{}}},\"providers\":{\"providersThrottleDuration\":2000000000,\"file\":{\"watch\":true,\"filename\":\"/etc/traefik/dynamic-conf.yml\"}},\"log\":{\"level\":\"DEBUG\",\"format\":\"common\"}}"

```

**Comment by ddtmachado** [comment]
@jrbalsas please note that while using `nc` you are keeping the connection alive, so it will never reach the timeout from the server.

To properly test that you will need a connection in time wait or not responding status.

**Comment by jrbalsas** [comment]
Traefik 1.7 with similar configuration work as expected using nc

**Comment by dduportal** [comment]
Hi @jrbalsas , could you provide a full reproduction case for both cases so we be sure that we analyze the same context as yours (like a docker-compose file) please?

**Comment by jrbalsas** [comment]
## traefik v1.7 lab
* **docker-compose.yml**
```yml
version: '2.2'
services:
  traefik:
    image: traefik:1.7
    command:
      - "--debug"
    ports:
      - "80:80"
    volumes:
      - ./traefik.toml:/etc/traefik/traefik.toml
  wwwserver:
    image: containous/whoami
```
* **traefik.toml**
```toml
[respondingTimeouts]
idleTimeout = "5s"
readTimeout = "10s"

[file]

[frontends]
  [frontends.frontend1]
    entryPoints = ["http"]
    backend = "wwwservers"

[backends]
  [backends.wwwservers]
          [backends.wwwservers.servers.server1]
                url = "http://wwwserver:80" 
```
### Tests v1
```bash
$ time printf "GET / HTTP/1.1\nHost: localhost\n\n" |  nc -C localhost 80
HTTP/1.1 200 OK
Content-Length: 333
...
real    0m5,012s
user    0m0,007s
sys     0m0,002s
```
`IdleTimeout` works as expected and finish idle connection after configured time.
```bash
$ time nc localhost 80
```
`readTimeout`, in docker desktop for mac (last stable update) the inactive connection ends after 10 seconds as expected, but in docker community edition (ubuntu, current stable update) process must be manually finished, i.e. Ctrl+C. I realize that this result has surprised me...

## traefik v2 lab
* **docker-compose.yml**
```yml
version: '2.2'
services:
  traefik:
    image: traefik:2.0
    ports:
      - "80:80"
    volumes:
      - ./traefik.yml:/etc/traefik/traefik.yml                                                                        
      - ./dynamic-conf.yml:/etc/traefik/dynamic-conf.yml  
  wwwserver:
    image: containous/whoami
```
* **traefik.yml**
```yml
log:
  level: DEBUG
providers:
  file:
    filename: /etc/traefik/dynamic-conf.yml
entrypoints:
  web:
    address: ":80"
    transport:
      respondingTimeouts:
        readTimeout: "10s"
        idleTimeout: 5
```
### Tests v2
```bash
$ time printf "GET / HTTP/1.1\nHost: localhost\n\n" |  nc -C localhost 80
HTTP/1.1 200 OK
Content-Length: 333
...
(connection must be manually finished, i.e. Ctrl+c)
```
`IdleTimeout` does not finish the idle request as configured
```bash
$ time nc localhost 80
```
`readTimeout`, inactive connection is not automatically finished after timeout not in docker desktop for mac (last stable update) nor docker community edition (ubuntu, current stable update).

# Conclusion
I just need an effective way to prevent inactive connections/idle requests that consume resources unnecessarily, i.e. for preventing slow http attacks, but I can not do this (or do not know how to)  by adjusting the entrypoints timeouts

**Comment by traefiker** [comment]
Closed by #6115.
